### PR TITLE
Add support for webhook subgroup events

### PR DESCRIPTION
**Pull request by vntw**
Docs are 100% up-to-date with the current payload: https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#subgroup-events. I also updated all links to the other webhook events since they have been moved.

I tried to look at existing capitalizations of SubGroup and align it with those, hope it's okay

### CODE REVIEW

File: event_parsing.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ const (
 	EventTypeSystemHook    EventType = "System Hook"
 	EventTypeTagPush       EventType = "Tag Push Hook"
 	EventTypeWikiPage      EventType = "Wiki Page Hook"
+	EventTypeSubGroup      EventType = "Subgroup Hook"
```
Comment (svanharmelen) [inline event_parsing.go:45]: Please order these alphabetically...

File: event_parsing.go [hunk 2]
```diff
@@ -269,6 +270,8 @@ func ParseWebhook(eventType EventType, payload []byte) (event interface{}, err e
 		event = &TagEvent{}
 	case EventTypeWikiPage:
 		event = &WikiPageEvent{}
+	case EventTypeSubGroup:
+		event = &SubGroupEvent{}
```
Comment (svanharmelen) [inline event_parsing.go:274]: Please order these alphabetically...

File: event_parsing_webhook_test.go [hunk 3]
```diff
@@ -450,3 +450,21 @@ func TestParseWikiPageHook(t *testing.T) {
...
+
+	event, ok := parsedEvent.(*SubGroupEvent)
+	if !ok {
+		t.Errorf("Expected SubGroupEvent, but parsing produced %T", parsedEvent)
+	}
+
+	if event.EventName != "subgroup_create" {
+		t.Errorf("EventName is %v, want %v", event.EventName, "subgroup_create")
+	}
+}
```
Comment (svanharmelen) [inline event_parsing_webhook_test.go:470]: Please order this function alphabetically among the other functions...

File: event_webhook_types.go [hunk 4]
```diff
@@ -962,3 +962,21 @@ type WikiPageEvent struct {
...
+	EventName      string `json:"event_name"`
+	Name           string `json:"name"`
+	Path           string `json:"path"`
+	FullPath       string `json:"full_path"`
+	GroupID        int    `json:"group_id"`
+	ParentGroupID  int    `json:"parent_group_id"`
+	ParentName     string `json:"parent_name"`
+	ParentPath     string `json:"parent_path"`
+	ParentFullPath string `json:"parent_full_path"`
+}
```
Comment (svanharmelen) [inline event_webhook_types.go:982]: Please order this struct alphabetically among the other structs...

File: event_webhook_types_test.go [hunk 5]
```diff
@@ -463,3 +463,30 @@ func TestReleaseEventUnmarshal(t *testing.T) {
...
+	}
+
+	if event.GroupID != 2 {
+		t.Errorf("GroupID is %v, want %v", event.GroupID, 2)
+	}
+
+	if event.ParentGroupID != 1 {
+		t.Errorf("ParentGroupID is %v, want %v", event.ParentGroupID, 1)
+	}
+}
```
Comment (svanharmelen) [inline event_webhook_types_test.go:492]: Please order this function alphabetically among the other functions...

File: event_webhook_types.go [hunk 6]
```diff
@@ -962,3 +962,21 @@ type WikiPageEvent struct {
...
 	} `json:"object_attributes"`
 }
+
+// SubGroupEvent represents a subgroup event.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#subgroup-events
+type SubGroupEvent struct {
+	CreatedAt      string `json:"created_at"`
+	UpdatedAt      string `json:"updated_at"`
```
Comment (svanharmelen) [inline event_webhook_types.go:972]: Both of these should be `*time.Time`

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: event_webhook_types.go (hunk 6 above)
Comment (vntw) [reply event_webhook_types.go]: Thanks, wasn't sure what to use since both variants are currently used - changed

**Review by svanharmelen — APPROVED** [review]
Thanks @vntw 👍🏻

**Pull request merged**